src/modules/data/BlockHeader.ts

```diff
@@ -120,4 +121,17 @@ export class BlockHeader
 
         return this;
     }
+
+    /**
+     * Collects data to create a hash.
+     * @param buffer - The buffer where collected data is stored
+     */
+    public computeHash (buffer: SmartBuffer)
```

Comment:
Why is this needed? We already have the hash in the header, why does stoa have to generate it again?

Reply:
We have the hash value of the previous block.
I added this function to calculate the hash value of the current block and input it into the database.

Reply:
Right you are correct.

We could have changed Agora to push a (Hash, Block) pair. But since you already did all this work then let's use your code.

Reply:
The problem that I see is that every time we update the hashing mechanism in Agora we will also have to update it in Stoa. It means doubling the amount of necessary work.

Reply:
Without this, we do not know the hash value of the last block entered.
I think it is necessary to develop other services such as Block Explorer.

Reply:
Understandable. Thanks.